hedera-node/hedera-mono-service/src/main/java/com/hedera/services/context/MutableStateChildren.java

```diff
@@ -60,7 +59,7 @@ public class MutableStateChildren implements StateChildren {
     private WeakReference<MerkleScheduledTransactions> schedules;
     private WeakReference<VirtualMap<VirtualBlobKey, VirtualBlobValue>> storage;
     private WeakReference<VirtualMap<ContractKey, IterableContractValue>> contractStorage;
-    private WeakReference<MerkleMap<EntityNumPair, MerkleTokenRelStatus>> tokenAssociations;
+    private NonAtomicReference<TokenRelStorageAdapter> tokenAssociations;
```

Comment:
Out of curiosity, what does the `NonAtomicReference` class accomplish?

Reply:
`NonAtomicReference` keeps the `TokenRelStorageAdapter` instance from being garbage-collected (since, unlike the `MerkleMap` and `VirtualMap` children, the `ServicesState` does not hold a reference to the `TokenRelStorageAdapter` child).